Adding more GraphQL documentation (#2637)
OrchardCMS/OrchardCore: src/OrchardCore.Modules/OrchardCore.Contents/README.md

    contentItemId: "417qsjrgv97e74wvp149h4da53"
  }) {
    contentItemId
    displayText
    publishedUtc
  }
}
```

### Available fields

These fields are available at the content item level:

| Property | 
| -------- | 
| `contentItemId` |
| `contentItemVersionId` |
| `contentType` |
| `displayText` |
| `published` |
| `latest` |
| `modifiedUtc` |
| `publishedUtc` |
| `createdUtc` |
| `owner` |
| `author` |

In addition, all the content parts can also be retrieved like this:

```graphql
{
  blogPost {
    displayText
    autoroutePart {
      path
    }
  }
}
```

### Query arguments

Different types of query arguments can be composed to filter the results:

- Ordering: Sorting content items by any field value using `orderBy`
- Filtering: Selecting content items in a query by scalar or relational filters using `where`
- Pagination: Slicing content items in a query using `first` and `skip`

#### Ordering

When querying all content items of a type you can supply the `orderBy` argument for every scalar field of the type: `orderBy: { <field>: ASC }` or  `orderBy: { <field>: DESC }`.

Order the list of all `BlogPost` content items ascending by `displayText`:

```graphql
query {

>   blogPost(orderBy: { displayText }) {

Reviewer: ```suggestion
  blogPost(orderBy: { displayText: ASC }) {
```

    contentItemId
    displayText
    publishedUtc
  }
}
```

> Order the list of all `BlogPost` content items descending by `publishedUtc`:

Reviewer: ```suggestion
Order the list of all `BlogPost` content items descending by `publishedUtc` and then ascending by `displayText`:
```

```graphql
query {

>   blogPost(orderBy: { publishedUtc: DESC }) {

Reviewer: ```suggestion
  blogPost(orderBy: { publishedUtc: DESC, displayText: ASC  }) {
```

    contentItemId
    displayText
    publishedUtc
  }
}
```

The field you are ordering by does not have to be selected in the actual query.

> It's also not possible to order responses by multiple fields or by related fields.

Reviewer: ```suggestion
It's also not currently possible to order responses by their parts or custom fields.
```

#### Filtering

When querying all content items of a type you can supply different parameters to the where argument to constrain the data in the response according to your requirements. 

> The available options depend on the scalar and relational fields defined on the type in question.

Reviewer: ```suggestion
The available options depend on the scalar and part fields defined on the type in question.
```

    contentItemId
    displayText
    publishedUtc
  }
}
```

##### Filtering by a concrete value

The easiest way to filter a query response is by supplying a concrete value for a certain field to filter by.

Query all `BlogPost` content items with a specific display text:

```graphql
query {
  blogPost(where: {
    displayText: "About"
  }) {
    contentItemId
  }
}
```

##### Advanced filter criteria

Depending on the type of the field you want to filter by, you have access to different advanced criteria you can use to filter your query response.

Query all `BlogPost` content items whose `displayText` is in a given list of strings:

```graphql
query {
  blogPost(where: {
    displayText_in: ["My biggest Adventure", "My latest Hobbies"]
  }) {
    contentItemId
    displayText
    publishedUtc
  }
}
```

Query all `BlogPost` content items whose creation date is less than a specific date:

```graphql
query {
  blogPost(where: {
    creationgUtc_lt: "2011-11-13T07:45:00"
  }) {
    contentItemId
    displayText
    publishedUtc
  }
}
```

##### Content Part Filters

For content parts, you can define conditions on the part by nesting the according argument in `where`.

> Query all `BlostPost` content items where the `autoroutePart` has a specific value in its `path`:

Reviewer: ```suggestion
Query all `BlogPost` content items where the `autoroutePart` has a specific value in its `path`:
```

For an `AND`-filter to evaluate to `true`, all of the nested conditions have to be `true`.
For an `OR`-filter to evaluate to `true`, at least one of the nested conditions has to be `true`.
For a `NOT`-filter to evaluate to `true`, all of the nested conditions have to be `false`.
Using `OR`, `AND` and `NOT`

Let's start with an easy example:

Query all `BlogPost` content items that are created in `2018` and whose `displayText` is in a given list of strings:

```graphql
query {
  blogPost(where: {
    AND: {
      displayText_in: ["My biggest Adventure", "My latest Hobbies"]
    }, {
      publishedUtc_gt: "2018"
    }
  }) {
    contentItemId
    displayText
    publishedUtc
  }
}
```

##### Arbitrary combination of filters with `AND`, `OR` and `NOT`

You can combine and even nest the filter combinators `AND`, `OR` and `NOT` to create arbitrary logical combinations of filter conditions.

Query all `BlogPost` content items that are created in `2018` and whose `displayText` is in a given list of strings, or have the specific `contentItemId` we supply:

```graphql
query {
  blogPost(where: {
    OR: {
      AND: {

>         displayText_in: ["My biggest Adventure", "My latest Hobbies"]

Reviewer: ```suggestion
        displayText_in: ["My biggest Adventure", "My latest Hobbies"],
```

>       }, {

Reviewer: ```suggestion
```
